cardano_node_tests/tests/test_plutus_mint_build.py

```diff
@@ -1117,3 +1118,90 @@ def test_past_horizon(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment_addrs: List[clusterlib.AddressRecord],
+    ):
+        """Test minting a token passing a redeemer for a simple minting script.
+
+        Expect failure.
+
+        * fund the token issuer and create a UTxO for collateral
+        * try to mint the token using a simple script passing a redeemer
+        * check that the minting failed because is expected a Plutus script
+        """
+        temp_template = common.get_test_id(cluster)
+        payment_addr = payment_addrs[0]
+        issuer_addr = payment_addrs[1]
+
+        lovelace_amount = 2_000_000
+        token_amount = 5
+        script_fund = 200_000_000
+
+        minting_cost = plutus_common.compute_cost(
+            execution_cost=plutus_common.MINTING_COST,
+            protocol_params=cluster.get_protocol_params(),
+        )
+
+        # Fund the token issuer and create UTXO for collaterals
+
+        mint_utxos, collateral_utxos, __ = _fund_issuer(
+            cluster_obj=cluster,
+            temp_template=temp_template,
+            payment_addr=payment_addr,
+            issuer_addr=issuer_addr,
+            minting_cost=minting_cost,
+            amount=script_fund,
+        )
+
+        # Create simple script
+        keyhash = cluster.get_payment_vkey_hash(issuer_addr.vkey_file)
+        script_content = {"keyHash": keyhash, "type": "sig"}
+        script = Path(f"{temp_template}.script")
+
+        with open(script, "w", encoding="utf-8") as out_json:
+            json.dump(script_content, out_json)
+
+        # Mint the "qacoin"
+        policyid = cluster.get_policyid(script)
+        asset_name = f"qacoin{clusterlib.get_rand_str(4)}".encode("utf-8").hex()
+        token = f"{policyid}.{asset_name}"
+
+        mint_txouts = [
+            clusterlib.TxOut(address=issuer_addr.address, amount=token_amount, coin=token)
+        ]
+
+        plutus_mint_data = [
+            clusterlib.Mint(
+                txouts=mint_txouts,
+                script_file=script,
+                collaterals=collateral_utxos,
+                redeemer_file=plutus_common.REDEEMER_42,
+            )
+        ]
+
+        tx_files = clusterlib.TxFiles(
+            signing_key_files=[issuer_addr.skey_file],
+        )
+        txouts = [
+            clusterlib.TxOut(address=issuer_addr.address, amount=lovelace_amount),
+            *mint_txouts,
+        ]
+
+        with pytest.raises(clusterlib.CLIError) as excinfo:
+            cluster.build_tx(
+                src_address=payment_addr.address,
+                tx_name=f"{temp_template}_step2",
+                tx_files=tx_files,
+                txins=mint_utxos,
+                txouts=txouts,
+                mint=plutus_mint_data,
+            )
+        assert (
+            "expected a script in the Plutus script language, but it is actually "
+            "using SimpleScriptLanguage SimpleScriptV1" in str(excinfo.value)
+        )
```

Comment:
I recently started doing following in tests like this:
```
err_str = str(excinfo.value)
assert "expected error string" in err_str, err_str
```
This way we see the whole error message in case the assertion fails, not a shortened representation of `excinfo` object.